homeassistant/components/keyboard_remote/__init__.py

```diff
@@ -266,12 +275,21 @@ async def async_device_start_monitoring(self, dev):
             _LOGGER.debug("Keyboard async_device_start_monitoring, %s", dev.name)
             if self.monitor_task is None:
                 self.dev = dev
+                # set the descriptor to the one provided to the config if any, falling back to the device path if not set
+                if self.config_descriptor:
+                    self.descriptor = self.config_descriptor
+                else:
+                    self.descriptor = self.dev.path
+
                 self.monitor_task = self.hass.async_create_task(
-                    self.async_monitor_input(dev)
+                    self.async_device_monitor_input()
                 )
                 self.hass.bus.async_fire(
                     KEYBOARD_REMOTE_CONNECTED,
-                    {DEVICE_DESCRIPTOR: dev.path, DEVICE_NAME: dev.name},
+                    {
+                        DEVICE_DESCRIPTOR: self.descriptor,
```

Comment:
Would it be interesting to include the device path too, besides the descriptor, in the event data?

Often they would be the same, but for links they could differ, right?

Reply:
I can't see a case where the device path, when being used with a link, would be useful. Its will be `/dev/input/eventN` where `N` will be assigned by the OS can can change on every boot, or if the device is re-plugged. As it won't be known in advance by the user, I don't see how anyone would filter events for it.

In the case where the full path is used in the config, then the path is still sent in the event as the descriptor.